src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,14 +34,22 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        String email = authentication.getName();
+        ShoppingCart cart = shoppingCartService.getByUser(
+                userService.findByEmail(authentication.getName())
+                        .orElseThrow(() -> new RuntimeException(
+                                "There is no user with such email: " + email)));
```

Comment:
you duplicate logic here, it would be better to create a private method, that gets user by the authentication